### PR TITLE
Fine-tune usage of dotcom-ui-base-styles package

### PR DESCRIPTION
**Pull request by andygout**
To be merged into https://github.com/Financial-Times/dotcom-page-kit/compare/matth/base-style-package.

Applies few finishing touches to work in that branch to enable:
- `dotcom-ui-layout` to consume `dotcom-ui-base-styles` for its styles.
- `dotcom-ui-shell` to consume `dotcom-ui-base-styles` for its fonts.

It also removes existing code relating to fonts that exists in `dotcom-ui-layout` (obsolete now it lives in `dotcom-ui-base-styles`).

### CODE REVIEW

File: packages/dotcom-ui-base-styles/styles.scss [hunk 1]
```diff
@@ -1,3 +1,5 @@
+$system-code: 'page-kit-base-styles' !default;
+
 @import 'n-ui-foundations/main';
```
Comment (i-like-robots) [inline packages/dotcom-ui-base-styles/styles.scss]: I reckon let's use `@use` here 😄

Comment (i-like-robots) [reply packages/dotcom-ui-base-styles/styles.scss]: And a big TODO comment about eventually sorting out `n-ui-foundations`

File: packages/dotcom-ui-shell/src/components/Shell.tsx [hunk 2]
```diff
@@ -37,17 +38,14 @@ function Shell(props: TShellProps) {
     ...props.scripts,
     ...props.resourceHints,
     // TODO: abstract font URLs into 'core branding' package
```
Comment (i-like-robots) [inline packages/dotcom-ui-shell/src/components/Shell.tsx]: This comment can go

File: packages/dotcom-ui-shell/src/components/Shell.tsx [hunk 3]
```diff
@@ -77,6 +75,7 @@ function Shell(props: TShellProps) {
         <Content contents={props.contents || props.children} />
         <AppContextEmbed appContext={props.appContext} />
         <FlagsEmbed flags={props.flags} />
+        <LoadFontsEmbed />
```
Comment (i-like-robots) [inline packages/dotcom-ui-shell/src/components/Shell.tsx]: This needs to be in the `<head>`

File: packages/dotcom-ui-base-styles/styles.scss (hunk 1 above)
Comment (andygout) [reply packages/dotcom-ui-base-styles/styles.scss]: Fixed in [this commit](https://github.com/Financial-Times/dotcom-page-kit/pull/769/commits/2a306f27e202aa446adda2f941fb796243f59fe7) and [this commit](https://github.com/Financial-Times/dotcom-page-kit/pull/769/commits/e24e547b096aec8aec6e5aa3bd9235cdc93dd1ec) respectively.

File: packages/dotcom-ui-shell/src/components/Shell.tsx (hunk 2 above)
Comment (andygout) [reply packages/dotcom-ui-shell/src/components/Shell.tsx]: Fixed in [this commit](https://github.com/Financial-Times/dotcom-page-kit/pull/769/commits/f7cd27e948aadfb06b03d912c54cb80f62db4f7e).

File: packages/dotcom-ui-shell/src/components/Shell.tsx (hunk 3 above)
Comment (andygout) [reply packages/dotcom-ui-shell/src/components/Shell.tsx]: Fixed in [this commit](https://github.com/Financial-Times/dotcom-page-kit/pull/769/commits/db8d9e2fabc5992941898f142f357c63f3649f7b).

**Pull request merged**